packages/material-ui/src/Avatar/Avatar.d.ts

```diff
@@ -5,7 +5,7 @@ declare const Avatar: OverridableComponent<{
   props: {
     alt?: string;
     childrenClassName?: string;
-    imgProps?: React.HtmlHTMLAttributes<HTMLImageElement>;
+    imgProps?: React.ImgHTMLAttributes<HTMLImageElement>;
```

Comment:
What is the difference between those types? Could you share an example that is valid at runtime but is rejected by this type?

Reply:
The difference is the HTML element that they refer to. The original is for the `<html />` element, and the change, is for the `<img />` element. 

Because they are for completely different elements, using [HtmlHTMLAttributes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a05cc538a42243c632f054e42eab483ebf1560ab/types/react/index.d.ts#L1904) and [ImgHTMLAttributes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a05cc538a42243c632f054e42eab483ebf1560ab/types/react/index.d.ts#L1926) interchangeably produce some weird errors. 

For instance, the following is not valid:

```ts
<Avatar imgProps={{ alt: "Lorem", height: 80, width: 80 }} />
```

Even though `{ alt: "Lorem", height: 80, width: 80 }` is a completely valid implementation of [ImgHTMLAttributes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a05cc538a42243c632f054e42eab483ebf1560ab/types/react/index.d.ts#L1926).

Attempting to write the above throws the following error:

```
Type '{ alt: string; height: number; width: number; }' is not assignable to type 'HtmlHTMLAttributes<HTMLImageElement>'.
  Object literal may only specify known properties, and 'alt' does not exist in type 'HtmlHTMLAttributes<HTMLImageElement>'.ts(2322)
  Object literal may only specify known properties, and 'height' does not exist in type 'HtmlHTMLAttributes<HTMLImageElement>'.ts(2322)
```

Reply:
Ah I didn't spot the double `HtmlHTML`. I thought we were already generic enough and the typings infer the props from the element. Makes perfect sense, thanks.